File: app/src/main/java/github/xtvj/cleanx/utils/RootShellRunner.java
// SPDX-License-Identifier: GPL-3.0-or-later

package github.xtvj.cleanx.utils;

import androidx.annotation.NonNull;
import androidx.annotation.WorkerThread;

import com.topjohnwu.superuser.Shell;

import java.io.InputStream;

class RootShellRunner extends Runner {
    @WorkerThread
    @NonNull
    @Override
    synchronized public Result runCommand() {
        Shell.Job shell = Shell.su(commands.toArray(new String[0]));
        for (InputStream is : inputStreams) {
            shell.add(is);
        }
        Shell.Result result = shell.exec();
        clear();
        return new Result(result.getOut(), result.getErr(), result.getCode());
    }
}

File: app/src/main/java/github/xtvj/cleanx/shell/RootShellRunner.kt
// SPDX-License-Identifier: GPL-3.0-or-later
package github.xtvj.cleanx.shell

import androidx.annotation.WorkerThread
import com.topjohnwu.superuser.Shell

internal class RootShellRunner : Runner() {
    @WorkerThread
    @Synchronized
    override fun runCommand(): Result {
        val shell: Shell.Job = Shell.su(*commands.toTypedArray())
        for (input in inputStreams) {
            shell.add(input)
        }
        val result = shell.exec()
        return Result(result.out, result.err, result.code)
    }

}